File: python3/7.py
class Solution:
    def reverse(self, x: int) -> int:
        if x < 0:
            y = int('-'+str(x)[:0:-1])
        else:
            y = int(str(x)[::-1])
        return y if y<=2147483647 and y>=-2147483648 else 0

File: c/7.c
class Solution:
    def magicTower(self, nums: List[int]) -> int:
        heap = []
        hp = 1
        ans = 0
        if sum(nums) < 0:
            return -1
        for num in nums:
            if num < 0:
                heapq.heappush(heap,num)
            hp += num
            while hp <= 0:
                hp -= heapq.heappop(heap)
                ans += 1
        return ans